client/state/plugins/installed/status/reducer.js

```diff
@@ -54,6 +55,29 @@ export default function status( state = {}, action ) {
 		case PLUGIN_INSTALL_REQUEST_FAILURE:
 		case PLUGIN_REMOVE_REQUEST_FAILURE:
 			return Object.assign( {}, state, { [ siteId ]: statusForSite( state[ siteId ], action ) } );
+		case PLUGIN_NOTICES_REMOVE: {
+			if ( ! action.statuses || ! action.statuses.length ) {
+				return state;
+			}
+
+			const allStatuses = Object.entries( state )
+				.map( ( [ stateSiteId, siteStatuses ] ) => {
+					const updatedSiteStatuses = Object.entries( siteStatuses ).filter(
+						( [ , pluginStatus ] ) => {
+							return ! action.statuses.includes( pluginStatus.status );
+						}
+					);
+
+					if ( ! updatedSiteStatuses.length ) {
+						return [];
+					}
+
+					return [ stateSiteId, Object.fromEntries( updatedSiteStatuses ) ];
+				} )
+				.filter( ( siteStatus ) => siteStatus.length );
+
+			return { ...Object.fromEntries( allStatuses ) };
```

Comment:
Is it necessary to spread this object?

Reply:
Good catch 👍  Removed in https://github.com/Automattic/wp-calypso/pull/48000/commits/93fede2a54b8cbde7cfb9582d99000b3c56fe1a5